idea-plugin/src/com/reason/bs/BscProcess.java

```diff
@@ -0,0 +1,62 @@
+package com.reason.bs;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.ColoredProcessHandler;
+import com.intellij.execution.process.OSProcessHandler;
+import com.intellij.execution.process.ProcessListener;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.reason.ORProcessException;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class BscProcess {
+
+  private final Project m_project;
+
+  public static BscProcess getInstance(@NotNull Project project) {
+    return ServiceManager.getService(project, BscProcess.class);
+  }
+
+  public BscProcess(Project project) {
+    m_project = project;
+  }
+
+  @Nullable
+  public Integer run(
+      VirtualFile sourceFile, List<String> arguments, ProcessListener processListener)
+      throws ORProcessException {
+    Optional<VirtualFile> bscPath = BsPlatform.findBscExecutable(m_project, sourceFile);
+    if (!bscPath.isPresent()) {
+      throw new ORProcessException("Unable to find bsc.exe.");
```

Comment:
I prefer to avoid exception here, and use warn log.
exceptions are very intrusive in the UI, and here we have a situation we can handle.